src/main/java/io/gingersnapproject/k8s/CacheRuleInformer.java

```diff
@@ -0,0 +1,224 @@
+      }
+
+      String ruleName = configuration.configMapName().get();
+      log.info("Informer on rules in {}", ruleName);
+      final long RESYNC_PERIOD = 60 * 1000L;
+      KubernetesClient kc = client.get();
+      informer = kc.configMaps().withName(ruleName).inform(new ConfigMapEventHandler(drm), RESYNC_PERIOD);
+      informer.start();
+   }
+
+   public void stop(@Observes ShutdownEvent ignore) {
+      if (informer != null) {
+         log.info("Shutdown informer");
+         informer.close();
+      }
+   }
+
+   private static final class ConfigMapEventHandler implements ResourceEventHandler<ConfigMap> {
+      private DynamicRuleManagement drm;
+      private interface SendEventFunc {
+         void sendEvent(String name, EagerCacheRuleSpec rule);
+      }
+
+      public ConfigMapEventHandler(DynamicRuleManagement drm) {
+         this.drm = drm;
+      }
+
+      private void processConfigMapAndSend(Set<Map.Entry<String, String>> entries, SendEventFunc f) {
+         for (Entry<String, String> entry : entries) {
+            Builder eRuleBuilder = EagerCacheRuleSpec.newBuilder();
+            try {
+               JsonFormat.parser().ignoringUnknownFields().merge(entry.getValue(), eRuleBuilder);
+            } catch (InvalidProtocolBufferException e) {
+               log.error("Cannot parse eager rule with name {}", entry.getKey(), e);
+               log.debug("{}", e);
+            }
+            f.sendEvent(entry.getKey(), eRuleBuilder.build());
+         }
+      }
+
+      @Override
+      public void onAdd(ConfigMap obj) {
+         processConfigMapAndSend(obj.getData().entrySet(), (name, rule) -> {
+            log.info("calling DynamicRuleManagement.addRule({},{}", name, rule.toString());
+         });
+      }
+
+      @Override
+      public void onUpdate(ConfigMap oldObj, ConfigMap newObj) {
+         Map<String, String> oldMap = oldObj.getData();
+         Map<String, String> newMap = newObj.getData();
+         Set<Map.Entry<String, String>> olds = oldMap.entrySet();
+         Set<Map.Entry<String, String>> news = newMap.entrySet();
+         Set<Map.Entry<String, String>> added, removed, changed;
+         added = new HashSet<Map.Entry<String, String>>(news);
+         // Get added keys. From new set remove
+         // - keys already present in old set
+         added.removeIf(arg0 -> {
+            return oldMap.containsKey(arg0.getKey());
+         });
+
+         removed = new HashSet<Map.Entry<String, String>>(olds);
+         // Get removed keys. From old set remove
+         // - keys still present in new set
+         removed.removeIf((arg0 -> {
+            return newMap.containsKey(arg0.getKey());
+         }));
+
+         changed = new HashSet<Map.Entry<String, String>>(olds);
+         // Get the changed keys. From the old set remove
+         // - keys which aren't in the new set (deleted)
+         // - keys which are in the new set with same value (unchanged)
+         changed.removeIf(
```

Comment:
All of this logic can be simplified still https://github.com/gingersnap-project/db-syncer/pull/42#discussion_r1039373166